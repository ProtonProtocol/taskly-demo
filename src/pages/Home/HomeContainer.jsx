import React from 'react';
import Home from './Home';
import ProtonSDK from '../../utils/proton';

class HomeContainer extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      windowWidth: 0,
      isLoggingIn: false
    }
  }

  componentDidMount() {
    this.updateWindowWidth();
    window.addEventListener('resize', this.updateWindowWidth)
  }

  updateWindowWidth = () => {
    this.setState({ windowWidth: window.innerWidth });
  }

  generateLoginRequest = async () =>  {
    const { setLoggedInState } = this.props;
    try {
      this.setState({ isLoggingIn: true });
      const { auth, accountData } = await ProtonSDK.login();
      setLoggedInState(auth.actor, auth.permission, accountData);
      this.setState({ isLoggingIn: false });
    } catch (e) {
      this.setState({ isLoggingIn: false });
      console.error(e);
    }
  }

  render() {
    const { windowWidth, isLoggingIn } = this.state;

    return(
      <Home openLoginModal={this.generateLoginRequest} windowWidth={windowWidth} isLoggingIn={isLoggingIn} />
    );
  }
}

export default HomeContainer;
