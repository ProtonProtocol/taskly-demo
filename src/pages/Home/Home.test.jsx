import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Home from './Home';

const createTestProps = props => ({
  openLoginModal: jest.fn(),
  windowWidth: 800,
  isLoggingIn: false,
  ...props,
});

describe('Home component', () => {
  let props;

  beforeEach(() => {
    props = createTestProps({});
  });

  it('renders without crashing', () => {
    shallow(<Home {...props} />);
  });

  it('matches the rendered snapshot with default props', () => {
    const component = shallow(<Home {...props} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('matches the rendered snapshot with isLoggingIn true and smaller window width', () => {
    props = createTestProps({
      windowWidth: 500,
      isLoggingIn: true
    });
    const component = shallow(<Home {...props} />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
