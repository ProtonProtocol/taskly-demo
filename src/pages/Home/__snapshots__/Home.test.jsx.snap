// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component matches the rendered snapshot with default props 1`] = `
<div
  className="page"
>
  <div
    className="page-wrapper"
  >
    <div
      className="header"
    >
      <img
        alt="Taskly"
        src="./images/taskly-logo.svg"
      />
      <div
        className="header-buttons"
      >
        <div
          className="header-button-item"
        >
          Services
        </div>
        <div
          className="header-button-item"
        >
          About Us
        </div>
        <div
          className="header-button-item"
        >
          Contact
        </div>
      </div>
    </div>
    <div
      className="main"
    >
      <h1>
        Get a virtual assistant. Organize like never before.
      </h1>
      <RedButton
        className="login-button"
        handleOnClick={[MockFunction]}
        isLoading={false}
        text="Login with Proton"
      />
      <div
        className="app-store-badges"
      >
        <img
          alt="Download on Google"
          src="./images/google-badge.png"
        />
        <img
          alt="Download on Apple"
          src="./images/apple-badge.png"
        />
      </div>
    </div>
    <RedBackground
      bottom="auto"
      left="auto"
      right="-20px"
      side="right"
      top="20%"
    >
      <img
        alt="Peeps!"
        src="./images/peeps.png"
      />
    </RedBackground>
  </div>
</div>
`;

exports[`Home component matches the rendered snapshot with isLoggingIn true and smaller window width 1`] = `
<div
  className="page"
>
  <div
    className="page-wrapper"
  >
    <div
      className="header"
    >
      <img
        alt="Taskly"
        src="./images/taskly-logo.svg"
      />
      <div
        className="header-buttons"
      >
        <HamburgerIcon />
      </div>
    </div>
    <div
      className="main"
    >
      <h1>
        Get a virtual assistant. Organize like never before.
      </h1>
      <RedButton
        className="login-button"
        handleOnClick={[MockFunction]}
        isLoading={true}
        text="Login with Proton"
      />
      <div
        className="app-store-badges"
      >
        <img
          alt="Download on Google"
          src="./images/google-badge.png"
        />
        <img
          alt="Download on Apple"
          src="./images/apple-badge.png"
        />
      </div>
    </div>
    <RedBackground
      bottom="auto"
      left="auto"
      right="-20px"
      side="right"
      top="20%"
    >
      <img
        alt="Peeps!"
        src="./images/peeps.png"
      />
    </RedBackground>
  </div>
</div>
`;
